packages/omg-js-childchain/src/rpc/rpcApi.js

```diff
@@ -13,46 +13,69 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */
 
-const fetch = require('node-fetch')
+const request = require('request-promise')
 const debug = require('debug')('omg.childchain.rpc')
 const JSONBigNumber = require('json-bigint')
 
 class RpcError extends Error {
-  constructor ({ code, description }) {
-    super(description || code)
-    this.code = code
+  constructor (data) {
+    super(data.description || data.code || data)
   }
 }
 
-async function get (url) {
-  return fetch(url).then(parseResponse)
+async function get ({ url, proxyUrl }) {
+  let fetch = request
+  if (proxyUrl) {
+    fetch = request.defaults({ proxy: proxyUrl })
+  }
+
+  try {
+    const res = await fetch.get(url)
+    return parseResponse(res)
+  } catch (err) {
+    throw new Error(err)
+  }
 }
 
-async function post (url, body) {
+async function post ({ url, body, proxyUrl }) {
   body.jsonrpc = body.jsonrpc || '2.0'
   body.id = body.id || 0
 
-  return fetch(url, {
-    method: 'POST',
-    headers: { 'Content-Type': 'application/json' },
-    body: JSON.stringify(body)
-  }).then(parseResponse)
+  let fetch = request
+  if (proxyUrl) {
+    fetch = request.defaults({ proxy: proxyUrl })
+  }
+
+  try {
+    const options = {
+      method: 'POST',
+      uri: url,
+      headers: { 'Content-Type': 'application/json' },
+      body,
+      json: true
```

Comment:
By setting `json: true` aren't you parsing the body into json, then stringifying it back later, then parsing it into json again? 😄 

Actually, now that we're using request-promise, it's got a `transform` option that you could use instead - it's essentially the same thing, but done all in one step.

I don't have a strong preference though, so it's up to you...

Reply:
@kevsul yea `json:true` just stringifies the body in the request

im not actually sure why we need to do...
// Need to use a JSON parser capable of handling uint256
    json = JSONBigNumber.parse(body)

in the response when the response comes back as an object already

Reply:
> when the response comes back as an object already

If we let the default json parser parse it, then it will handle big numbers (> Number.MAX_SAFE_INTEGER) incorrectly.
For example, if the response from the server is something like:
```
{"value": 123456789123456789}
```
it will lose precision and be returned as 
```
value: 123456789123456780 //  number
```

Ethereum uses 256 bit integers and it's common to see very large values for wei. 
So we need to explicitly parse the request body using JSONBigNumber.parse(), which puts number values into a BN object, i.e.
```
value: BigNumber // with a value of 123456789123456789
```

Now that I look at it again though, my original comment was off - I thought `json
true` was affecting the response not the request 🤦‍♂